.github/workflows/linux.yml

```diff
@@ -52,3 +52,10 @@ jobs:
           name: test logs
           path: |
             build/meson-logs/
+      - name: Upload doxygen docs
```

Comment:
the frustrating thing about github workflows is the amount of duplication required. There's no good/easy way (afaik) to compose a single image, then run just the build step on that image (like we do on gitlab with the freedesktop ci-templates). And workflows are separate too, so if we had a separate pages *workflow* we couldn't even re-use the package lists.

We *could* move the pages job into a separate `jobs` entry and then at least have the pip/apt list as an env var.

But even then it's a lot of duplication - this here is the easiest way to avoid that duplication.

And in case the "upload" term is confusing: this just stores the artifact from the workflow, it doesn't do the actual GitHub pages upload, that's handled in the other workflow which fetches this artifact and pushes it out to pages. And that one only runs on push-to-master.

We could add an extra [`if:` condition](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif) like this - just need to figure out the right magic variables:
```yaml
- name: Upload doxygen docs
  if: ${{ github.event_name == 'pull_request' && github.event.action == 'unassigned' }}
```
but... meh?